cranelift/codegen/src/context.rs

```diff
@@ -192,25 +167,11 @@ impl Context {
 
         self.remove_constant_phis(isa)?;
 
-        if let Some(backend) = isa.get_mach_backend() {
-            let result = backend.compile_function(&self.func, self.want_disasm)?;
-            let info = result.code_info();
-            self.mach_compile_result = Some(result);
-            Ok(info)
-        } else {
-            self.regalloc(isa)?;
-            self.prologue_epilogue(isa)?;
-            if opt_level == OptLevel::Speed || opt_level == OptLevel::SpeedAndSize {
-                self.redundant_reload_remover(isa)?;
-            }
-            if opt_level == OptLevel::SpeedAndSize {
-                self.shrink_instructions(isa)?;
-            }
-            let result = self.relax_branches(isa);
-
-            log::trace!("Compiled:\n{}", self.func.display(isa));
-            result
-        }
+        let backend = isa.get_mach_backend().expect("only mach backends nowadays");
```

Comment:
Can you add a FIXME to make this non-optional?

Reply:
```suggestion
        // FIXME: make this non optional
        let backend = isa.get_mach_backend().expect("only mach backends nowadays");
```

Reply:
More generally, I expect that we'll continue to tweak the data structures to remove `Option<>`s on new bits and delete the old bits; the CLIF function representation still has some bits that can go (`jt_offsets` for example?) too.

Reply:
I'd expect that too, yeah. The fact that so many things were marked `pub` (and not `pub(crate)`) make it so the compiler is not seeing all the dead code out there.